native/core/src/activities/object.rs

```diff
@@ -0,0 +1,145 @@
+use matrix_sdk::ruma::OwnedEventId;
+use urlencoding::encode;
+
+use crate::models::{ActerModel, AnyActerModel};
+
+#[derive(Clone, Debug)]
+pub enum ActivityObject {
+    News {
+        object_id: OwnedEventId,
+    },
+    Pin {
+        object_id: OwnedEventId,
+        title: String,
+    },
+    CalendarEvent {
+        object_id: OwnedEventId,
+        title: String,
+    },
+    TaskList {
+        object_id: OwnedEventId,
+        title: String,
+    },
+    Task {
+        tl_id: OwnedEventId,
+        object_id: OwnedEventId,
+        title: String,
+    },
+    Unknown {
+        object_id: OwnedEventId,
+    },
```

Comment:
Just with an additional case, where we might not have been able to load the item locally, so we just pass out the parent id